test/test_parse_osmchange_input.cpp

```diff
@@ -481,401 +255,221 @@ void test_node() {
+}
 
+TEST_CASE("Create relation, member ref too large", "[osmchange][relation][xml]") {
+  REQUIRE_THROWS_AS(process_testmsg(
+    R"(<osmChange><create><relation changeset="972" id="-1">
+           <member type="relation" ref="99999999999999999999999999999999" role="stop"/>
+           </relation></create></osmChange>)"), http::bad_request);
+}
 
-  // Member: role may have up to 255 unicode characters
-  for (int i = 1; i <= 256; i++) {
+TEST_CASE("Create relation, role with <= 255 unicode characters", "[osmchange][node][xml]") {
+  for (int i = 1; i <= 255; i++) {
     auto v = repeat("😎", i);
-
-    try {
-      process_testmsg(
-          fmt::format(
+    REQUIRE_NOTHROW(process_testmsg(
+      fmt::format(
                R"(<osmChange><create><relation changeset="858" id="-1">
                            <member type="node" role="{}" ref="123"/>
                   </relation></create></osmChange>)",
-           v));
-      if (i > 255)
-        throw std::runtime_error("test_relation::010: Expected exception for "
-                                 "string length > 255 unicode characters");
-    } catch (http::exception &e) {
-      if (e.code() != 400)
-        throw std::runtime_error("test_relation::010: Expected HTTP 400");
-      const std::string expected = "Relation Role has more than 255 unicode characters";
-      if (std::string(e.what()).substr(0, expected.size()) != expected)
-        throw std::runtime_error(
-            "test_relation::010: Expected Relation Role has more than "
-            "255 unicode characters, got: " + std::string(e.what()));
-      if (i <= 255)
-        throw std::runtime_error("test_relation::010: Unexpected exception for "
-                                 "string length <= 255 characters");
-    }
+           v)));
   }
 }
 
-void test_invalid_data() {
+TEST_CASE("Create relation, role with > 255 unicode characters", "[osmchange][node][xml]") {
+  REQUIRE_THROWS_MATCHES(process_testmsg(
+    fmt::format(
+               R"(<osmChange><create><relation changeset="858" id="-1">
+                           <member type="node" role="{}" ref="123"/>
+                  </relation></create></osmChange>)",
+           repeat("😎", 256))), 
+    http::bad_request, Catch::Message("Relation Role has more than 255 unicode characters at line 2, column 321"));
+}
 
-  std::string s = "\x3C\x00\x00\x00\x00\x0A\x01\x00";
-  try {
-    process_testmsg(s);
-    throw std::runtime_error("test_invalid_data::001: Expected exception");
-  } catch (http::exception &e) {
-    if (e.code() != 400)
-      throw std::runtime_error("test_invalid_data::001: Expected HTTP 400");
-  }
+
+// INVALID DATA TESTS
+
+TEST_CASE("Invalid data", "[osmchange][relation][xml]") {
+  REQUIRE_THROWS_AS(process_testmsg("\x3C\x00\x00\x00\x00\x0A\x01\x00"), http::bad_request);
 }
 
-void test_large_message() {
+
+// LARGE MESSAGE TESTS
+
+TEST_CASE("Very large XML message", "[osmchange][relation][xml]") {
```

Comment:
Very large XML message is testing nodes only, so maybe the tag should be `[osmchange][node][xml]`?